storage/internal/benchmarks/upload_benchmark.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+type uploadParams struct {
+	o              *storage.ObjectHandle
+	contentsReader io.ReadSeeker
+	readerLength   int
+	objectSize     int64
+	chunkSize      int
+}
+
+func uploadBenchmark(ctx context.Context, u uploadParams) (time.Duration, error) {
+	elapsedTime := time.Duration(0)
+	start := time.Now()
+
+	u.o.If(storage.Conditions{
+		DoesNotExist: true,
+	})
+
+	objectWriter := u.o.NewWriter(ctx)
+	objectWriter.ChunkSize = u.chunkSize
+
+	var written int64
+	for written < u.objectSize {
+		// stop timer here
+		elapsedTime += time.Since(start)
+		rewindReader(u, written)
```

Comment:
Can you speak to why pausing time here is necessary? Also again, I'm wondering if writing the data to a rewindable reader is really what we want to be measuring here.

Reply:
I paused time here to remove the time to rewind the reader (which admittedly is probably not that significant) from the benchmark, as that would not be a typical workload.

I used a rewindable reader here on a random string buffer to avoid reallocating a buffer. Perhaps that would be a more typical workflow?